presto-native-execution/src/test/java/com/facebook/presto/nativeworker/TestHiveWindowQueries.java

```diff
@@ -84,4 +85,105 @@ public void testRowNumber()
+                "RANGE CURRENT ROW",
+                "RANGE BETWEEN UNBOUNDED PRECEDING AND UNBOUNDED FOLLOWING",
+                "RANGE BETWEEN CURRENT ROW AND UNBOUNDED FOLLOWING",
+                "ROWS UNBOUNDED PRECEDING",
+                "ROWS CURRENT ROW",
+                "ROWS BETWEEN UNBOUNDED PRECEDING AND UNBOUNDED FOLLOWING",
+                "ROWS BETWEEN CURRENT ROW AND UNBOUNDED FOLLOWING");
+
+        for (String overClause : overClauses) {
+            for (String frameClause : frameClauses) {
+                StringBuilder windowFunctionCalls = new StringBuilder();
+                for (Pair<String, String> functionCall : functionCalls) {
+                    // Value functions require a total ordering of input rows to produce deterministic results with window frames in ROWS mode. The argument column for the value
+                    // function is therefore passed as a parameter in functionCall and included in the ORDER BY clause.
+                    windowFunctionCalls.append(String.format("%s OVER (%s, %s %s),", functionCall.first(), overClause, functionCall.second(), frameClause));
+                }
+                windowFunctionCalls.deleteCharAt(windowFunctionCalls.length() - 1);
+                queries.add(String.format("SELECT %s FROM lineitem", windowFunctionCalls.toString()));
+            }
+        }
+        return queries.build();
+    }
+
+    @Test
+    public void testNthValue()
+    {
+        List<Pair<String, String>> nthValueFunctionPairs = Arrays.asList(Pair.of("nth_value(receiptdate, 5)", "receiptdate"),
+                Pair.of("nth_value(partkey, 5)", "partkey"),
+                Pair.of("nth_value(shipmode, 5)", "shipmode"),
+                Pair.of("nth_value(tax, orderkey + 1)", "tax"),
+                Pair.of("nth_value(linenumber, orderkey + 1)", "linenumber"));
+
+        List<String> queries = buildWindowTestQueriesWithFrame(nthValueFunctionPairs);
+        for (String query : queries) {
+            assertQuery(query);
+        }
+    }
+
+    protected void testAggregateWindowFunction(String aggregateFunction, Boolean numericAggregate)
+    {
+        ImmutableList.Builder<Pair<String, String>> aggregateFunctionPairs = ImmutableList.builder();
+        aggregateFunctionPairs.add(Pair.of(String.format("%s(partkey)", aggregateFunction), "partkey"));
+
+        // Varchar and Date datatypes are tested with columns shipmode and receiptdate respectively for min, max and count aggregate functions.
+        if (!numericAggregate) {
+            aggregateFunctionPairs.add(Pair.of(String.format("%s(shipmode)", aggregateFunction), "shipmode"));
+            aggregateFunctionPairs.add(Pair.of(String.format("%s(receiptdate)", aggregateFunction), "receiptdate"));
+        }
+
+        List<String> queries = buildWindowTestQueriesWithFrame(aggregateFunctionPairs.build());
+        for (String query : queries) {
+            assertQuery(query);
+        }
+    }
+
+    @Test
```

Comment:
Move all the test methods together after the non-test helper methods.

Reply:
Done, thanks for the suggestion!